app/src/main/java/com/kabouzeid/gramophone/appshortcuts/AppShortcutIconGenerator.java

```diff
@@ -53,11 +54,16 @@ private static Icon generateThemedIcon(Context context, int iconId, int foregrou
         Drawable vectorDrawable = Util.getTintedVectorDrawable(context, iconId, foregroundColor);
         Drawable backgroundDrawable = Util.getTintedVectorDrawable(context, R.drawable.ic_app_shortcut_background, backgroundColor);
 
-        // Squash the two drawables together
-        LayerDrawable layerDrawable = new LayerDrawable(new Drawable[]{backgroundDrawable, vectorDrawable});
+        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.O){
+            AdaptiveIconDrawable adaptiveIconDrawable = new AdaptiveIconDrawable(backgroundDrawable, vectorDrawable);
+            return Icon.createWithAdaptiveBitmap(drawableToBitmap(adaptiveIconDrawable));
```

Comment:
It might be good to rely on `IconCompat`, actually.

Reply:
Just curious, where is the benefit in using IconCompat?

Reply:
The general idea of preferring compat libraries is for more consistent behaviour across API versions and the benefit of any future feature additions/bug fixes. It does come with the tradeoff of a slightly larger app though.